src/udev/udev-builtin-net_id.c

```diff
@@ -243,9 +243,29 @@ static bool is_pci_ari_enabled(sd_device *dev) {
         return streq(a, "1");
 }
 
+static bool is_pci_bridge(sd_device *dev) {
+        const char *v, *p;
+
+        if (sd_device_get_sysattr_value(dev, "modalias", &v) < 0)
+                return false;
+
+        if (!startswith(v, "pci:"))
+                return false;
+
+        p = strrchr(v, 's');
+        if (!p)
+                return false;
+        if (*++p != 'c')
+                return false;
+
+        /* PCI device subclass 04 corresponds to PCI bridge */
+        return streq(++p, "04");
```

Comment:
hmm, you could replace everything starting from the strrchr() with `returns endswith(v, "sc04")`, no?